pkg/kubelet/dockertools/manager.go

```diff
@@ -1861,7 +1972,94 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, _ api.PodStatus, podStatus *kubec
 		}
 	}
 
-	// Start everything
+	next, status, done := findActiveInitContainer(pod, podStatus)
```

Comment:
Yeah, PLEG will drive sync loop, so we'll start (then exit), then get
triggered by the complete from the PLEG, then record that we started and
start the next one, etc.  If we started synchronously we would block PLEG
events, so this is the more natural way within the current pod worker (it's
not adding latency, since PLEG gives us just in time invocation).

On Fri, Apr 8, 2016 at 4:41 PM, Prashanth B <notifications@github.com>
wrote:

> In pkg/kubelet/dockertools/manager.go
> <https://github.com/kubernetes/kubernetes/pull/23567#discussion_r59086268>
> :
>
> > @@ -1861,7 +1972,94 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, _ api.PodStatus, podStatus *kubec
> >  		}
> >  	}
> >
> > -	// Start everything
> > +	next, status, done := findActiveInitContainer(pod, podStatus)
>
> each invocation of SyncPod will only start one container, and there's a
> lag between invocations? Or did I miss a for loop somewhere here.
>
> I think this runs in a podworker goroutine, if that's the case we should
> just be able to wait for all init's to start/exit synchrounsly?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23567/files/21f0e955e3f1d05950fed76280c28d3e81d90742..ad7a96fe3b9dda804fee77e41f7e14ad0b4513d7#r59086268>
>